### DISCUSSION TITLE
[Feature]: preserve album info from upload client

### DISCUSSION THREAD

**Discussion by dorianim** [immich-app/immich#1678] [Feature Request]
### Feature detail

When uploading many photos which are already sorted into albums locally, it would be nice to have those albums copied to the server so that the photos don't have to be sorted again.

### Platform

Mobile App

**Comment by alextran1502** [comment]
This can be reword as backup and preserve album info from mobile device

**Reply by tvlpirb** [reply]
I believe this would be one of the most important features for mobile use case. Most mobile users have multiple albums locally, being able to sync all while preserving album info on initial and subsequent sync is expected behaviour. Otherwise the gallery becomes a disarray with the current behaviour. 

Would such a feature be easy to implement? Similar to how the CLI tool can choose which album to sync to.  If not, what are some of the issues with implementing it if someone were to make a PR?

**Reply by lynrayy** [reply]
>  subsequent sync is expected behaviour.

[tvlpirb](https://github.com/tvlpirb) Yes!

**Comment by zackpollard** [comment]
I'm not sure how easy this would be to do, might be possible on iOS, but on android the amount of different gallery and camera apps might make this problem far too vast.

**Reply by Cohinem** [reply]
It already show what the name of the album is and how many images there is. it only needs to create that album in Immich and sort them accordingly.

**Comment by dorianim** [comment]
Well, on Android, you select albums for backup and those albums should be preserved

**Comment by Txxicer** [comment]
Great feature, but not only in mobile device, if this function would be available in combination with Ticket [#34](https://github.com/immich-app/immich/issues/34) this would be great!

**Comment by mind-overflow** [comment]
definitely a +1 for this.  
  
it's super useful if e.g. you want to upload both camera pictures and generally less important media that gets sent in group chats, but you don't want the camera roll to get all mixed up with those images.  
it's currently impossible because even if you manually add all the uploaded pictures to the "camera roll" album, every time you take a new one you have to manually add it and that just gets tiring imho.

**Comment by MoldyTaint** [comment]
+1 to show support for this feature!

Sorry, new to the github thing. Didn't know it was a big issue. I'll change it to the thumb up thing!

**Reply by bo0tzz** [reply]
Please upvote the post with a 👍 instead of just commenting +1.

**Comment by tvlpirb** [comment]
What is the update on this feature?
For anyone else interested in this, a current workaround this issue is to make a script which uses the CLI tool and puts everything in an album using the --al option

**Reply by 2600box** [reply]
Could you explain a bit more detail about using your workaround method? Thanks!

**Reply by tvlpirb** [reply]
I've actually moved on and just settled with nextcloud with manual album tagging. However, previously what I did was sync the files to my backup server, the backup server would then periodically use the immich cli tool to upload all media to my immich server. I made use of the --al option which puts the media in an album on upload either based of a given name or a parent folders name. You can figure out the rest and experiment. 

If you're syncing the media to your server, only to then use the CLI tool to add to immich be wary that you're storing everything twice. So make sure to clean up automatically once its added to immich.

It might be possible to use the immich CLI directly from phone but I haven't attempted that.

**Comment by tvlpirb** [comment]
Suggestions to the UI/UX for this feature for anyone making a PR, if we had to implement this feature it would be nice to be able to choose which album a local one uploads to. In addition to this, the user should be able to choose whether it uploads to an album or not to maintain the current behaviour.

**Comment by p4block** [comment]
Without this I can't use Immich. I've imported using the CLI my albums  (DCIM from pics from 2010 and onwards, Reddit memes, WhatsApp images...) but my phone will blindly upload everything untagged, despite knowing well from which album each pic is from. 

Reasoning being that the merged timeline view is completely unusable, specific album views are needed to filter out the crap from groups etc and only view actual camera pics. 

I even tried to game the system by creating a folder with the same name as the local folder on the server, but that ofc doesn't work :laughing:

**Reply by klaus1k** [reply]
Same here - immich is so mature right now, I will use it in production right away once this feature is implemented.
I wish I could contribute a PR to help the team achieve this, but sadly I not a programmer :/

**Reply by Cohinem** [reply]
Yes exactly. almost everything was good till I discovered this issue. It's not useable for me at this state. I can't manually make albums for and sort tens of thousands of images.

**Comment by shant1010** [comment]
WOW this is such an important feature request! I can't add albums manually through the app either because it just brings up the entire Immich library.

**Comment by Jamminroot** [comment]
I unknowingly wrote a duplicate for the original post of the discussion.
Here it is:

TL;DR: I'd like to be able to divide my uploads into folders and\or automatically assign albums to them

I've got a bunch of various folders on my phone: camera roll, some photos imported from GoPro, some photos in a separate folder based on the occasion (like "Vacation" or "Holiday Trip" etc.), bunch of memes to show my wife, screenshots, pictures from my work and others.
I'd like all of them to be backed up and be accessible, but the problem is that all of them get mixed into one flat collection, thus causing all them to be show in the same collection.

Now with any new onboarding to using the application user would get thousands of assets, all mixed into one collection. Assigning albums manually would be a problem - for both already uploaded assets, or every time one downloads a new funny gif\picture.

Also worth mentioning that when I download tv series, memes, work images or anything non-related to actual photos I take, It'd be great to be able to ignore all those irrelevant assets in face recognition: that would allow to reduce the resource cost on the feature, and "Face roll" would not be swarmed with faces of random ppl \ images (imagine having a face of a dear one somewhere in the end of the list, after a ton of irrelevant persons :) )

The only alternative I can have atm is to ignore everything but actual camera roll.

Hope that makes sense, and I really hope that it would be considered, as it is quite a painful downside for such a great tool.

**Comment by traktuner** [comment]
+1

My duplicate for reference:

https://github.com/immich-app/immich/discussions/3932

**Comment by XRTce** [comment]
Just a bump on this feature as I believe it could be very useful to organize your pictures. The great majority is mainly managing their photos on their phone. So they likely already organized their assets into iOS or android albums thus preserving album information would be just the right thing to do. Regarding UI/UX I think it would be nice to manage this feature as opt in per album basis. For example I want all assets in my Birthday iOS album to be synced  to the My Birthday album on server. I will then go to the sync tab in the app, select album sync, turn on album sync using a list view which came up showing all my albums and a toggle next to them. Additionally I can tap on the album name and edit the name of the album to which the photos should be synced on the server side. If the album does not exist it will be created

**Reply by 2600box** [reply]
FWIW, this is one of the things holding me back and I guess it may be for many others too.

**Reply by Gecko-with-a-hat** [reply]
Yeah this is also the exact feature I am missing to start deploying Immich in my household.

**Comment by maslanypotwor** [comment]
This is a feature I am waiting for!
It would be the cherry on top that will push me to finally organize all my images (which are now a mess...) around Immich infrastructure.

**Comment by klaus1k** [comment]
Now that there is a storage template for {{albums}}, this feature is the last piece of the puzzle thats missing for me. The workflow should be working similar to the "album" flag of the CLI.

On mobile I select different albums right now: "Camera", "Lightroom", "WhatsApp", that I can backup to immich. Right now they do not adhere to the {{album}} storage template, because they are in no album for immich.
With that feature one could duplicate the folder structure AND the albums from mobile, which would be pretty awesome!!!

If that feature gets implement for external libraries as well, I soon am out of feature requests for immich! ;)

**Comment by DaanHend** [comment]
Definitely something that should be in. Would be great to have more variations inside the original photo structure divided by device & device 'albums' when I backup my devices.

As Immich is a backup tool first. I'm not sure why this has a "Nice to Have" tag though.

#5648

**Comment by driesmp** [comment]
Very useful if you have a folder structure already that can create albums automatically from a folder structure. One folder = One album

**Comment by robertocmochetti** [comment]
Today was my first time using Immich and I ended up on this thread because it was the only feature I wish I could find. Hope it gets implemented soon because everything else on the software is absolutely perfect for my user case.

**Reply by davidSGE** [reply]
Same story here. It's so important to me and should work easily on iOS and Android since you have to manually select the albums you want to upload anyway

**Reply by rage1337** [reply]
This is exactly my story, too.

Hey, to day is my birthday - please make me present and announce this feature!

**Comment by shant1010** [comment]
It’s funny, I keep getting emails from this thread. I would think it would have been a must have feature. On Jan 2, 2024, at 6:05 PM, robertocmochetti ***@***.***> wrote:﻿
Today was my first time using Immich and I ended up on this thread because it was the only feature I wish I could find. Hope it gets implemented soon because everything else on the software is absolutely perfect for my user case.

**Comment by aisbergde** [comment]
Interesting, that this features seams to be so hard to implement. OneDrive upload has it, Synology Drive has it.

Folders needs to be selected in the mobile client (Android) anyway, so the information should be there. This feature request is just about to use the source folder in the upload template.

For now, I have no other workaround, only then to upload the camera folder and the Screenshot folder, no other folders. Screenshots have the "Screenshots" inside the filename, I can use a database script, to mark all screenshots as archived. Somehow it works. But I can't save other folders this way because I don't want to mix.

**Comment by TheAnachronism** [comment]
This is also about just preserving an already existing folder structure. The same would be immensely useful for external libraries, as those are most often already structured in some kind of way.

**Comment by aisbergde** [comment]
if it is not possible to use the source folder in the template right now, it would be helpful to keep this information at least in the database for later using. Currently this important metadata will be lost.

Later usage could be:
- tag
- reorganize (migrate) to a new template, containing the source
- ...

**Comment by 2600box** [comment]
The most logical to my mind is taking album info and assigning them as tags. This allows for multiple "albums" without duplicating files.

> Later usage could be:
> 
> * tag

**Comment by Trusty555** [comment]
Im new to immich, seemed good as I explored it. But this is the biggest grip I have. I have many albums organised in windows folders. I assumed the import feature would import from the root folder and create albums of the sub-folders. But it just uploads them all without any separation. 

Heres how it should work, 2 options:

1. Automatically create a folder structure or albums based on the folders from the external libraries. /import/cats creates: cats album 
2. Add multiple external libraries that upload directly to a chosen album. external library name: Dogs creates Dogs album

and if you didnt want an album, you would just delete the album, leaving the media in the main library

**Reply by aisbergde** [reply]
why you need to import into the default library, instead of just using the concept of external libraries? Using external libraries, all your folder structure is preserved. If your current assets are not on the same device, then just copy the data, not using any import, and mount as external libraries.

**Comment by Symbiot78** [comment]
+1 for this feature.

**Comment by SpikedCola** [comment]
Like others, I came here after trying to figure out why the albums I have created on my phone hadn’t transferred to the server. 

Please implement this, otherwise I can’t use the album feature on the phone to sort photos. It will save having to re-sort thousands of photos on the server. 

I would support some kind of bounty to fix this. 

Thanks

**Comment by lihaqi** [comment]
Yes, I hope this feature can be implemented

**Comment by nirmal-mewada** [comment]
Similar but slightly different, If this can be combined with https://github.com/immich-app/immich/discussions/1687 to provide a generic way to "sync" local albums that can manage import and export, that would be very useful.

**Comment by maogongzi** [comment]
This is really a nice feature to have, I'm using Android, but for now I only choose the **Camera** album to be backup-ed, although I have some other local albums like "screenshots" which I would also like to backup. I used to backup some non-photo pictures, and later I have to manually sort them out one by one and add them to a new album, that's a huge load of time.

**Comment by Behem07h** [comment]
the app already lets us view images by album via the "Albums on Device" category of the "Library" tab, surely immich can at least use the  library page for adding images to albums?

as a new user trying to get into immich, having to manually re-sort all images i backup is a huge drawback. People looking for a backup solution likely have hundereds of images already, surely you cant expect them to sort them all? In the effort it takes me to do that, I could go through the rest of the popular image backup solutions and find one that does it's job better.

Baffled how this is only "nice to have" when it should have been a key feature from the get go. how do we have a machine learning face detector but not this.

**Comment by moritzthecat** [comment]
this is excellent project - thanx

did setup the storage template to include album info and hoped it will preserve the iOS album info I used for the backup. `{{album}}/{{y}}/{{MM}}/{{y}}-{{MM}}-{{dd}} {{HH}}-{{mm}}-{{ss}} {{filename}}`
but this looks still in discussion ... 

**agree with other posts above that this is much more than nice to have**

looking for a workflow to manage several iPhone photos
maybe a good idea to summarize intended use of immich and workflows as epics or use cases to discuss further, could be supported by voting

methods 
import and which data shall be preserved - albums / folders could be 1st step
digital twin which can be worked on both ends 
export and/or sync albums/folders to smartphone to avoid double sorting work 

todays smartphones easily store more than 10.000 photos
just uploading without respecting existing meta data such as iOS albums or folders makes the solution less UX friendly

looking for a feasible workflow that includes a feasible way back to sync result of work to smartphone, again keeping meta data

think this could become a killer feature

**Comment by Skywalker8510** [comment]
+1

my duplicate for reference 
#6627

**Comment by Konrni** [comment]
To bring the discussion a bit forward, what would be a good design choice to implement this?
Right now "CLI --album" does an album join after an upload. As far as i understand, this is triggered on every upload.
This looks like the way to go, so it's only called on new items and doesn't "lock" existing ones to an album.

UI wise i believe it should be a checkbox placed besides the album preview button on the select albums to backup page.

Extra:
Global additional "Album" in settings or beside "recent" (=all pictures)
Text-box instead of a checkbox. "Forced Album Name" would be stored locally. Could there be a way to store it on the server?

**Comment by Moosbee** [comment]
> Yes, we are working on refactoring the mobile app backup and have that particular feature in mind as well with the refactor
osorin on Discord

**Comment by lynrayy** [comment]
Bump. Still actual.

**Comment by shant1010** [comment]
Im happy to see that this thread is still alive! This team has done some
amazing work.

**Comment by arrio464** [comment]
cool! Can’t wait to see this feature implemented!
This is exactly what I'm currently missing while using it.

**Comment by shant1010** [comment]
No, I can’t believe it lolOn Mar 1, 2024, at 10:04 AM, FlyingT ***@***.***> wrote:﻿
Damn, is this still not a feature?

**Comment by ThesiiNCey** [comment]
is this feature no implimented yet ?

**Reply by lgrawet** [reply]
No, unfortunately. This prevents me from using immich for phone sync.

**Reply by lynrayy** [reply]
@lgrawet

Same

**Reply by arrio464** [reply]
same😭

**Reply by gionag** [reply]
same

**Comment by dev-alto** [comment]
This would definitely increase QoL and allow me to ditch my built-in Photos app.

**Comment by manishmaharjan** [comment]
Indeed one of the highly important feature. Hope this feature will be out soon!

**Comment by baldsealion** [comment]
Are you serious, after all the configuration and setup, I can't sync photos with an album in iOS? Literally a dealbreaker until fixed sadly.

**Reply by BadCactus634** [reply]
Same. I just can't have all the photos that I care of, blended together with all the crap from Whatsapp, screenshot, documents and downloads... that will overload my server, as well as my connection.
I'll try uploading only the photos from my camera roll, the consequence is that this way, Immich will just be another app on my mobile phone, will not replace the Gallery app, until fixed.

**Reply by manishmaharjan** [reply]
Well, we all are desperately waiting for these features. However, personally I would prefer this  https://github.com/immich-app/immich/discussions/5409 folder view feature to be out as that would automatically bring this feature also along with it.

**Comment by Mikaka27** [comment]
Is there a way to query immich api for each picture to get which album it belongs to? That way I could sort them automatically on the server (but in a different directory than immich's).

Edit: I mean, does the server know about the albums on the phone at all or not?

**Reply by mmomjian** [reply]
The server doesn't know anything about the albums on the phone.

**Reply by Mikaka27** [reply]
Thanks for information :)

**Comment by 2600box** [comment]
Maybe I should wait longer, to see this feature implemented, but I am thinking of mass applying captions to photos stored in iCloud which I could then manage in immich as a proxy for albums.

Has anyone tried this as a workaround?

**Comment by lynrayy** [comment]
Still no albums on server?

P.s BUMP

**Reply by bo0tzz** [reply]
Please don't pointlessly bump feature requests, it doesn't accomplish anything.

**Comment by lynrayy** [comment]
Is there any forks with storing albums as folders on server?

**Reply by beercity** [reply]
Likely isnt something that can be accomplished with a fork, unless the fork maintainer also develops and releases a mobile app that has this feature as well. Unlikely that would happen. Will need to wait for Immich to rework the mobile upload feature.

**Comment by juliendargelos** [comment]
I also really miss this feature, I can't backup my library if it means loosing all my albums. I would say it is the most important thing for Immich to have regarding backup.

I can't sort thousands of photos twice, yet I would be okay manually mapping each album selected for backup to an album on Immich. I feel like this approach could be implemented without disrupting the existing backup system too much.

**Comment by lynrayy** [comment]
Immich devs just don't care...
Pigs

**Reply by juliendargelos** [reply]
if it makes you that mad, you should contribute to make the feature exist, not insult the people working for free on an open source software.

**Reply by bo0tzz** [reply]
@lynrayy you've already been toeing the line with impatience and this is just straight up abusive. The only reason I'm not removing it is so that others can see the amount of thumbs-downs indicating how unacceptable this is. We've previously expressed that we're open to this feature but we haven't gotten around to building it ourselves, as mentioned above any PRs are always welcome so if you want this feature that badly I'd recommend you channel your energy that way instead.

**Reply by jrasm91** [reply]
@juliendargelos  For the record, some of us _are_ getting paid to work on this now (since 1.5 months ago) :tada: , but your point still stands :smile:

**Reply by juliendargelos** [reply]
great!

**Comment by ghost** [comment]
Question to maintainers: how hard is it to implement this feature? Might find time to look into it, but need to know why it hasn't been looked into in 2 years.

**Reply by alextran1502** [reply]
It's harder than it looks. We need to decide how to handle cases where multiple albums are selected as backup. Right now, we take a unique list of assets from those albums to upload. You will need a mechanism to indicate which albums those assets are from.

I started to work on this and then got caught up with other tasks

**Reply by Moosbee** [reply]
Lets think this through, (I'm on a train excuse mistakes):
**Assumptions**:
The Phone has a looong list of all images on the phone.
One image can exist in 0, 1, 2 or more phone album.
One immich image can exist in 0, 1, 2 or more immich albums.

This would seem simple, upload image and add it to the corresponding albums, creating ones wich haven't.

I assume the difficult part would be to see what album should be added to the image( we may not want the camera album, but we do want the WhatsApp album) and to what immich album we need to add them(we have WhatsApp album for two different phones).

In case I'm wrong(wich I assume) please correct me

**Reply by alextran1502** [reply]
The other difficult part is that the timeline currently shows the assets selected for backup. So we will have to decouple what is shown on the timeline vs what will be backed up to make this mechanism simpler

**Reply by XRTce** [reply]
I don't think it should be an issue to show all local assets in the timeline because we have the indicator in the bottom right to show if the asset is backed up or not. Maybe add an "included albums for timeline view" setting so people can still decide which albums should be shown. Then we can default to matching the album names in Immich and the phone. If some time is left, adding the possibillity to override the default mappingof immich <> phone album so we could solve some problems coming from different platforms e.g. ios whatsapp pictures is called "WhatsApp" and Android "WhatsApp Images". The matching table containing immich album id and local album name should be stored on the phone itself or maybe inside the database but then tied to the device ID

**Comment by SpikedCola** [comment]
Anyone else suspicious that, due to the popularity of this feature request, it is going to become one of their "licensed-only" features?

**Reply by alextran1502** [reply]
@SpikedCola nope, as stated, There will never be paywalled features 😁 No matter how hard you try to convince us 🤭

**Reply by SpikedCola** [reply]
I sincerely hope this stays true.

**Comment by leuchtrakete666** [comment]
A good practice could be, to get an option in the app, to which album on the server you want the album on phone be matched to. I have no problem to find all the pics in the timeline meshed up. If you than don't want the photos in the timeline, you can always choose all the photos in the album and put them to the archive. 
WhatsApp phone albums and other instant messaging albums are especially a pain. I get so much bullshit there. It would be really cool, if they would be on separate albums on the server, where I than can sort out what I don't want to have. Sorting out on the phone would be an option of course too, but a very very uncomfortable option.
My workaround for this now is to search for filename patterns. Like Whatsapp "WA00" or Messenger "att."

**Reply by lynrayy** [reply]
Nah. It is better to do automatized album matching. Manual album syncing already exists in syncthing. This is not very convenient

**Reply by XRTce** [reply]
Maybe this is helpful: https://github.com/immich-app/immich/discussions/1678#discussioncomment-10118298

**Comment by Cohinem** [comment]
This really still not implemented? It all went good till I discovered this issue. I can't use Immich with this. tens of albums with tens of thousands of images. I can't manually sort all of these. And I can't have them randomly uploaded as it's too much of a mess and makes looking for something specific impossible.

**Comment by DrieWielRr** [comment]
Me too found out after syncing my mobile phone that everything got dumped in a single container without the possibily to differenciate where the data originated from. 
For me thats not a dealbreaker but for the wifey with pictures everywhere that she wants backupped its not an option, because it'll mix camera/whatsapp/instagram/etc. in one big pool and there is no way to untangle it all.
I'm using the storage-template and was rooting for {{album}} tag to fix at least the disklocation but found out it does absolutely nothing and still dumps media from different albums together in one big pile of pictures.

So at the moment there is no way for her to fully start using Immich which is a shame because its the most perfect fit for her needs.


Possible workarround:
- Have the mobile-app upload the album tag(s) within the meta-data so the server is aware of the album where the picture originated so we can search on the album tag ("whatsapp" for example) and then manually transfer all found items to a album within Immich.
I'ts not perfect but at least it gives us the means to bulk-resort the pictures within Immich. 

- A great addition would also be to at least have the {{album}} tag fixed for the storage-template, this way at least the data will be somewhat sorted on the physical disk in case something happens to Immich and backups are also scrap..
I'm currently using "{{album}}/{{y}}/{{MM}}/{{assetId}}" but on disk I still only see "/library/b1ba3686-262d-46ef-b5c9-27b1e5cce4d0/2024/01/83664fe3-cb97-4c75-8ac9-55213e0db9e2.jpg" while I expected to see "/library/b1ba3686-262d-46ef-b5c9-27b1e5cce4d0/**camera**/2024/01/83664fe3-cb97-4c75-8ac9-55213e0db9e2.jpg"

**Comment by lynrayy** [comment]
A lot of people want this feature. Why no any reaction from devs?

**Reply by 2600box** [reply]
they have, the most recent being less than a month ago: https://github.com/immich-app/immich/discussions/1678#discussioncomment-10115678

**Reply by DrieWielRr** [reply]
It seems they are trying to implement a full-fletched solution (but it has been 2+ years sind the request was made) and we still don't even have a viable work arround. If one has 25K pictures spread over multiple libraries on their phones (my wife has around 40K atm) there has to be a way to at least sort them manually within a couple of minutes without having to process every single picture/video by hand..

I mean dont get me wrong, im not complaining because I already think its a great piece of software but then again auto-sorting must be one if not the most important thing you can have within a media library..
I simply cant imagen anyone wanting personal pictures mixed with random whatsapp images and other crap that does require a backup but should be sepperated on the server (from a database perspective but also be sortable on the file level).
Just pretend that your database goes kaput and you find out that your backups are compromised as well, have fun trying to sort 100k of pictures.
It would have been a giant step forward if the server adds the physical location of the image at the time of syncing to the metadata so that we can write scripts to manually add pictures to albums or at least in a worst-case scenario are able to write a sorting script that reads the meta-data and can put images back to a understandable sorted file-structure on the disk.

**Reply by lynrayy** [reply]
> im not complaining

But i AM complaining

**Reply by bo0tzz** [reply]
@lynrayy the only thing you will accomplish with this repeated complaining is that we eventually get tired of it and block you from commenting under the immich-app org. Please stop.

**Reply by Cohinem** [reply]
Hey @bo0tzz , I wonder if there is any plan to add this feature? It's a great software you all are making. I have it installed on my server and mobile and update it regularly hoping for this feature to come so I can backup my media. Thanks <3.

**Comment by alextran1502** [comment]
@Cohinem A POC has been opened, you can follow here #11902 for the technical discussion

**Comment by DrieWielRr** [comment]
@bo0tzz 
It does make me wonder how the devs are handling this, I'm not asking for a full-fletched implementation I'm just looking for work arounds.
Like I said I think this is an amazing piece of software and it's definitely in the top-tier list of photo library software available.
But (ofcourse there is a but) I must assume the devs are using the software as well themselves and wonder how they got around this limitation of losing track where which picture belongs within the thousands of different pictures that are saved.
I want to fully implement Immich but my wife just doesn't want to migrate from her phone/onedrive because the prospect of losing any order in heer ~50K pictures gives her a depression just thinking about it.... 😂

So are there any known workarounds?
manual queries on the database + custom scripting that can put stuff back in order after its synced?
I mean there must be something I can do here to put her mind at easy that I can fix her albums after everything has been transferred?

**Reply by bo0tzz** [reply]
> this limitation of losing track where which picture belongs within the thousands of different pictures that are saved.

Many people are not doing in-depth asset organisation in their phone gallery, and so for them this simply isn't an issue.

**Reply by alextran1502** [reply]
TL;DR, we know you want it; it is a more complex mechanism, and we are working on other tasks, but we will get there


This feature is more complex to implement because the application allows users to select multiple albums for inclusion and exclusion. We have to factor in those use cases, and we are also tied to other work. 

Regarding your questions, I belong to the group of people who don't often create albums on my phone. Everything can stay in the timeline. I use the timeline for browsing and the search mechanism when I want to search for a specific photo, or I manage the album directly in Immich.

We strive to build quality software, so we don't often just put a dirty workaround into the current implementation; plus, with the current implementation, it is hard to put in a workaround and ensure everyone can benefit from it.

**Reply by DrieWielRr** [reply]
> TL;DR, we know you want it; it is a more complex mechanism, and we are working on other tasks, but we will get there
> 
> This feature is more complex to implement because the application allows users to select multiple albums for inclusion and exclusion. We have to factor in those use cases, and we are also tied to other work.
> 
> Regarding your questions, I belong to the group of people who don't often create albums on my phone. Everything can stay in the timeline. I use the timeline for browsing and the search mechanism when I want to search for a specific photo, or I manage the album directly in Immich.
> 
> We strive to build quality software, so we don't often just put a dirty workaround into the current implementation; plus, with the current implementation, it is hard to put in a workaround and ensure everyone can benefit from it.


I see your point, thanks for the replies! :)

**Reply by aisbergde** [reply]
We see the point. Maybe most immich developers are males.

Our wives and daughters ask us: Why should we switch from Google Photos to immich if there are no advantages in their eyes? The transfer to Chromecast doesn't work with immich (how are they supposed to present their photos?), they don't care where and how the data is stored (Google or my NAS). We have to offer them something that Google doesn't. We can't take anything away that Google offers. And as long as dad can't show us what's better than Google, dad is welcome to play around with immich, but he shouldn't bother the rest of the family with it. We need added value for our women.

**Reply by alextran1502** [reply]
Ladies and gents, this is a feature I want to work on, just busy with other features and tasks at the moment, haha. We are not dismissing this feature

**Reply by Behem07h** [reply]
while I acknowledge that that is the case, I had previously asked if there were any way I could hack together a temporary solution using external scripting, and was effectively told that even if it were possible, I would receive no help in my ventures, not a lick of information, because someone was "already working on implementing it".
If everyone is so busy working on other features, what is so wrong with helping people try to hack together a script that does what they need it to?

**Reply by alextran1502** [reply]
@Behem07h I think the wrong is using hacks, which can create tech debt.

**Reply by Behem07h** [reply]
i specified that i would be using external scripts to achieve it. There is no tech debt to be had if everything done is outside of the app and repo

**Reply by Behem07h** [reply]
a script to achieve this needs to:

    receive image and album data from the local photo gallery
    upload them with the api call
    check albums and create if missing using api calls
    assign the relevant images to the albums (again using api calls)

since it is an external script, it does not need to implement any of the other functionality that the official app has. it only needs to be able to upload and move images

**Reply by alextran1502** [reply]
@Behem07h I think the feature you want is for external library while this thread is for the mobile client 😅

**Reply by Behem07h** [reply]
no, i am in the right thread.
this is about the mobile client not preserving the album data of the phone's local photo gallery when images are uploaded, yes?
Since the api to upload images, create albums and move images into albums already exists (we can do all this manually in the mobile client after all), then it is trivial to use the same api to create a script that does the upload, album creation and album assignment automatically.

**Reply by jrasm91** [reply]
If you can copy the files off of your phone you can use an external tool to upload them to immich and create albums at the same time.

**Reply by Behem07h** [reply]
yes, thats exactly my point. but instead i was told to just give up trying and wait for an official implementation

**Reply by bo0tzz** [reply]
@Behem07h I can't find where this past discussion happened, so I don't know how it went exactly, but in general we tend to not spend much of our time helping people develop their own custom workarounds when that time is better spent working on Immich itself. We're not at all opposed to you creating hacks, but that doesn't mean we should drop things to guide you through it ;)

**Comment by metheoryt** [comment]
I came up with a quite simple solution, how to sort media by albums in first sync (means your immich is a fresh install and has no other media):

- Select single album for backup in immich mobile app
- Backup it
- Select all non-archived media in immich web app
- Create an album with the same name as on mobile, and add all selected media to it
- Archive whole album
- Repeat for other albums
- Unarchive media from all albums

However, this is not applicable for media, that is stored in multiple albums - only one album will be preserved for those

**Comment by VladoPortos** [comment]
Oh man, this is bit of a deal breaker for us :( Wanted to move out from nextcloud cause it was struggling with a lot of pictures, but what we used most was the backup feature from phone... 

Simply select folder like "Camera" and back everything into "Camera" album, or better yet let us specify the Album name.  Instead dumping all into main album...

**Reply by ThesiiNCey** [reply]
It has already been confirmed in the discord that this feature is coming, but without a date for it. I think if we continue to be annoying, it will be pushed further and further back

**Reply by Behem07h** [reply]
yeah we should just stop complaining and trust that the feature is coming soon(tm)...

the devs are working on it, they're just busy with other more important features, like ai image search, map stuff and person identification. those are definitely more important than the very first feature any onboarded user interacts with. 

how else will people find all their images containing food, or all the images they took in some holiday location? its not like they were already in neatly organised *albums* or anything, who would do that?

**Reply by DrieWielRr** [reply]
I dont think its being "annoying" but more like giving feedback. 
The devs already stated in a message that they thought this feature was not important enough compared to other features because they dont use it/need it, the community on the other hand has been sending them feedback by commenting that it is an important feature and why they think it is. 
Calling that annoying is rather odd imho (and purposly pushing it back would be rather childish, imo nobody would do that), they have created a masterpiece of software but many-many people cannot implement it because it lacks certain (imo "basic" features).
It sounds more like an outcry for attentention because many really wants to start using this software (me included) but simply cant because it will create a mess of our libraries. 
So thats a lot of people that might even be willing to send them some coffee or pizza (i know i would) for the hard work they have been putting in to this project that are now sidelined.

**Reply by 2600box** [reply]
I think the devs have received the message loud and clear and they are very open about the status of this project.
![Screenshot 2024-08-23 at 10 28 39](https://github.com/user-attachments/assets/c8e5d223-133d-4614-9c4d-4cbb60fb7cc5)

**Reply by Mikaka27** [reply]
> yeah we should just stop complaining and trust that the feature is coming soon(tm)...
> 
> the devs are working on it, they're just busy with other more important features, like ai image search, map stuff and person identification. those are definitely more important than the very first feature any onboarded user interacts with.
> 
> how else will people find all their images containing food, or all the images they took in some holiday location? its not like they were already in neatly organised _albums_ or anything, who would do that?

How can you demand that feature from the devs? This is a free software, you're not a customer.

**Reply by Behem07h** [reply]
thats an interesting way to interpret my statement. where do I ever *demand* for anything? 
in fact, where in that message do i even *request* anything?

**Reply by bo0tzz** [reply]
You are complaining that our priorities are different from yours, which comes off as a bit entitled. There are many features still missing from Immich, and every request thread has someone saying it's an essential feature that is preventing them from using Immich and how is it possible that it's not implemented yet! It will come, just be patient.

**Reply by alextran1502** [reply]
I am working on this #11965. Will probably make it to the next release

**Comment by alextran1502** [comment]
It is implemented in #11965. The feature will be available in release `v1.113.0`

**Reply by tvlpirb** [reply]
Thank you so much for your hard work! I know it's a lot of effort working on this for free and with everyone demanding different features, it must be tiresome.

I hope others can donate to the project and keep it going.

**Reply by Cohinem** [reply]
Thanks for the great work <3

**Reply by Cohinem** [reply]
@tvlpirb Just to note. I'm not sure if Alex is one of them. but AFAIK, some of them are getting paid to work on this project. They are being funded by FUTO.

**Reply by ghost** [reply]
Thanks @alextran1502 :heart:

**Reply by dmiales** [reply]
Well, finally. This was the only problem when switching from NextCloud

**Reply by Mikaka27** [reply]
Will I need to reupload all photos? Or will existing uploaded photos also sync to their albums?

**Reply by alextran1502** [reply]
@Mikaka27 you don't need to reupload, I added a button to sync uploaded photos

**Reply by Timi7007** [reply]
That's great and brings us one step closer to a full Google Photos importer, thank you!

**Reply by Gecko-with-a-hat** [reply]
> @Mikaka27 you don't need to reupload, I added a button to sync uploaded photos

Amazing!! I am really happy with this update. I can not wait to use it. 
I do have 1 more question, does this mean that the album info on the Immich server gets updated when I move a picture on my phone from album A to album B (local albums on my Samsung gallery)?

**Reply by alextran1502** [reply]
@Gecko-with-a-hat it won't be moved automatically, but if you pressed the sync button it will put the newly added photo in album B on your phone gallery to album B on Immich, it won't remove the photo from album A though

**Reply by sleepy-cloud** [reply]
Thank you, I really need this feature!

**Reply by VladoPortos** [reply]
Thanks so much ! we finally switched thanks to this feature :)

**Comment by BirgerK** [comment]
Hi Peepz,
I'm currently very interested into this feature. But I can't find the buttons in my iOS-app.
How to sync album-info from iOS-app to immich-server?

Thanks and Cheers!

**Reply by bo0tzz** [reply]
It's in the backup album selection settings.

**Comment by Tomagori** [comment]
Not sure this is the most appropriate place for this, but there is an issue (at least with the android app) where this doesn't *fully* replicate folder structures. E.g. "Whatsapp Images" has subfolders (usually "Sent" and "Private"), but Immich throws all images, including in subfolders, together under the general album name "Whatsapp Images". Thus, you lose the ability to differentiate what images were received and which were sent. 

It would be great if Immich preserved subfolders as well, though I guess at that point we veer into the discussion around nested album structures too.
